torch/_inductor/codegen/block_analysis.py

```diff
@@ -167,7 +171,11 @@ def match_affine_block_expr(
         stride.
         """
         index = cls._preprocess(index)
-        stride = sympy.Wild("stride", exclude=[index_var])
+        stride = sympy.Wild(
+            "stride",
+            exclude=[index_var],
+            properties=[lambda x: x.is_integer and x.is_nonnegative],
```

Comment:
nit: I think you can list multiple properties instead of using a lambda.
```suggestion
            properties=[x.is_integer, x.is_nonnegative],
```